coremltools/converters/mil/frontend/torch/test/test_torch_ops.py

```diff
@@ -503,6 +503,20 @@ def test_number_norm(self, shape, backend, p, keepdim):
             TorchBaseTest.run_compare_torch(shape, model, backend=backend, places=2)
 
 
+class TestWeightNorm(TorchBaseTest):
+    @pytest.mark.parametrize(
+        "in_features, out_features, backend",
+        itertools.product(
```

Comment:
Great test,
I would suggest to add more modules for the testing converge for different rank of weight,
for intance, the conv3d module would have rank 5 weight, and conv2d / 1d would have rank 4/ 3 respectively. This would converage more cases then just changing the shape of a 2d weight for the linear layer

Reply:
Done. These tests are taking some time to run so I've only added simple/short test for conv2d and conv3d.

To run all the `TestWeightNorm` currently takes about 35 seconds.

Reply:
I have the suggestion that deduce the combination of the linear model test,
so for instance, we can just hard code only a few in/out features number:
`in_out_features = [(1, 1), (2, 10), (20, 10)]`
Currently we are testing 9 combination with 5 dim in each test, so totally 45 tests.
I think I would rather make the linear test with 3 combination of in/out feture, so totally 3 * 5 tests,
and make the conv2/con3 test more robust by going through all possible axes (7 + 9 more tests)

Reply:
Sounds good. Done.

I think if we want to test negative dimension too, then `conv2` has 9 models and `conv3` has 11 models.